kits/arm/example_AR_kit.cpp

```diff
@@ -0,0 +1,200 @@
+        mobile -> sendText("Softstart Complete!");
+        softstart = false; 
+        continue;
+      }
+      arm -> send();
+
+      // Stay in softstart, don't do any other behavior
+      continue;
+    } 
+
+    // Get latest mobile_state
+    auto mobile_state = mobile->getState();
+    MobileIODiff diff(last_mobile_state, mobile_state);
+
+    // Get latest gripper_state
+    // gripper -> getState();
+
+    // Button B1 - Return to home position
+    if (diff.get(1) == MobileIODiff::ButtonState::ToOn) {
+        ar_mode = false;
+        arm -> setGoal(arm::Goal::createFromPosition(4, home_position));
+    }
+
+    // Button B3 - Start AR Control
+    if (diff.get(3) == MobileIODiff::ButtonState::ToOn) {
+      xyz_phone_init << mobile -> getLastFeedback().mobile().arPosition().get().getX(),
+                        mobile -> getLastFeedback().mobile().arPosition().get().getY(),
+                        mobile -> getLastFeedback().mobile().arPosition().get().getZ();
+      std::cout << xyz_phone_init << std::endl;
+      rot_phone_init = makeRotationMatrix(mobile -> getLastFeedback().mobile().arOrientation().get());
+      ar_mode = true;
+    }
+
+    // Button B6 - Grav Comp Mode
+    if (diff.get(6) == experimental::MobileIODiff::ButtonState::ToOn) {
+      arm -> cancelGoal();
+      ar_mode = false;
+    }
+
+    // Button B8 - End Demo
+    if (diff.get(8) == experimental::MobileIODiff::ButtonState::ToOn) {
+      // Clear MobileIO text
+      mobile -> clearText();
+      return 1;
+    }
+
+    if (ar_mode) {
+      // Get the latest mobile position and orientation
+      Eigen::Vector3d phone_pos;
+      phone_pos << mobile -> getLastFeedback().mobile().arPosition().get().getX(),
+                   mobile -> getLastFeedback().mobile().arPosition().get().getY(),
+                   mobile -> getLastFeedback().mobile().arPosition().get().getZ();
+      auto phone_ori = mobile -> getLastFeedback().mobile().arOrientation().get();
+
+      // Calculate rotation matrix from orientation quaternion
+      auto rot_phone_target = makeRotationMatrix(phone_ori);
+
+      // Calculate new targets
+      Eigen::Vector3d xyz_scale;
+      xyz_scale << 1, 1, 2;
+      Eigen::Matrix3d rot_target = rot_phone_init.transpose() * rot_phone_target * rot_init_target;
+      Eigen::Vector3d xyz_target = xyz_home + (0.75 * xyz_scale.array() *
+                                (rot_phone_init.transpose() * (phone_pos - xyz_phone_init)).array()).matrix();
+
+      // Calculate new arm joint angle targets
+      target_joints = arm -> solveIK(arm -> lastFeedback().getPosition(),
+                                              xyz_target, 
+                                              rot_target);
+
+      // Create and send new goal to the arm
+      arm -> setGoal(arm::Goal::createFromPosition(target_joints));
+
+      // double open_val = (mobile_state.getAxis(3) * 2)-1; // makes this range between -2 and 1
+      // gripper -> update(open_val);
```

Comment:
And this can just be done with an arm->setAuxState

(I'm fine leaving as is for the initial commit, though, given uncertainties around gripper API)